90zfsbootmenu/lib/zfsbootmenu-core.sh

```diff
@@ -1221,17 +1224,13 @@ has_resume_device() {
 # -m, -r, -e are print in the order they are passed into the function
 
 timed_prompt() {
-  local prompt delay message
-
-  delay="30"
-  prompt="Press $( colorize green "[RETURN]") or wait $( colorize yellow "%0.${#delay}d" ) seconds to continue"
-  message=()
+  local prompt delay message opt OPTIND
 
-  local opt OPTIND
   while getopts "d:p:m:r:e:" opt; do
     case "${opt}" in
       d)
         delay="${OPTARG}"
+        [ "${delay}" -gt 0 ] || return 0
```

Comment:
If `$delay` is not numeric, this test will fail and 0 will be returned.
- What's the right behavior if the delay argument is bogus: using the default 30-second timer or dying with no prompt?
- If the answer is "dying with no prompt", is passing a bogus time more like pressing `RETURN` or `ESCAPE`?

If it's better to revert to the default 30-second timer when delay is bogus, it should be sufficient to switch to
```sh
[ "${delay}" -le 0 ] && return
```
If it's better to die but we want a bogus time to look like `ESCAPE` while a valid non-positive numeric time looks more like `RETURN`, consider the following:
```sh
if ! [ "${delay}" -gt 0 ]; then
  [ "${delay}" -le 0 ] || return 1
  return 0
fi
```
Any numeric comparison fails when the argument is not numeric, so moving the negation away from the operator means we do a single comparison in the most common case when `$delay` is a positive value.

Reply:
I opted to solve this in a slight different fashion, which I think covers all of the bases. I'm not dead-set on this course, but I think it offers the best compromise of validating user input vs time spent in `timed_prompt`. Thoughts?